installers/rpm-code/codes/srw.sh

```diff
@@ -60,16 +49,47 @@ index 9e052df..0ab17fb 100644
 EOF
 }
 
-srw_python_patch_uti_math_eigen() {
-    patch --quiet env/work/srw_python/uti_math_eigen.py <<'EOF'
-@@ -29,7 +29,7 @@
- #from sklearn.decomposition import PCA as PCA #OC04082021 (moved to try-except)
- try:
-     from scipy.linalg import eigh as largest_eigh
--    from scipy.sparse.linalg.eigen.arpack import eigsh as largest_eigsh
-+    from scipy.sparse.linalg import eigsh as largest_eigsh
- except:
-     print("UtiMathEigen WARNING: SciPy unavailable; to install: pip install scipy")
- try:
+srw_srwpy_backwards_compatible() {
+    declare -a old_modules=(
+        srwl_bl
+        srwl_uti_brightness
+        srwl_uti_cryst
+        srwl_uti_mag
+        srwl_uti_smp
+        srwl_uti_smp_rnd_obj2d
+        srwl_uti_smp_rnd_obj3d
+        srwl_uti_src
+        srwl_uti_und
+        srwlib
+        srwlpy
+        uti_io
+        uti_io_genesis
+        uti_mag
+        uti_math
+        uti_math_eigen
+        uti_parse
+        uti_plot
+        uti_plot_com
+        uti_plot_matplotlib
+    )
+
+    declare d=$(codes_python_lib_dir)
+    install -m 444 /dev/stdin $d/srwpy_import_warning.py <<EOF
+displayed = False
+
+def check(calling_module):
+    global displayed
+    if not displayed:
+        displayed = True
+        print(f"This method of importing {calling_module} is deprecated. Please change to import from srwpy (ex: import {calling_module} -> from srwpy import {calling_module})")
```

Comment:
add `file=sys.stderr` arg to print so stdout isn't corrupted by warning.

Reply:
8c0f8d7fbcbe9043ee23049f39fca0719946fe1a